app/assets/javascripts/web_device_interface.ts

```diff
@@ -14,16 +14,23 @@ export class WebDeviceInterface extends DeviceInterface {
       timeout || setTimeout.bind(getGlobalScope()),
       setInterval.bind(getGlobalScope())
     );
-    this.database = new Database();
   }
 
   setApplication(application: SNApplication) {
-    this.database.setAlertService(application.alertService!);
+    const database = new Database(application.identifier, application.alertService!);
+    this.databases.push(database);
+  }
+
+  private databaseForIdentifier(identifier: ApplicationIdentifier) {
+    return this.databases.find(database => database.databaseName === identifier)!;
   }
 
   deinit() {
     super.deinit();
-    this.database.deinit();
+    for(const database of this.databases) {
+      database.deinit();
+    }
```

Comment:
So does this mean DeviceInterface only gets deinited once for _all_ applications? If it's still on an app-by-app basis I assume this would wreck every app instance.